### ISSUE TITLE
Query on VIEW with OFFSET NULL and WHERE condition involving CURRENT_USER() unexpectedly fetches a row

### ISSUE THREAD

**Issue by mrigger** [cockroachdb/cockroach#44203]
Consider the following test case:

```sql
CREATE TABLE t0(c0 BOOL);
INSERT INTO t0(c0) VALUES (false);
CREATE VIEW v0(c0) AS SELECT c0 FROM t0 WHERE t0.c0 OFFSET NULL;
SELECT * FROM v0 WHERE CURRENT_USER() != ''; -- unexpected: a row is fetched
```

The row should not be fetched, which is also the observed behavior when the `WHERE` clause is omitted:
```sql
SELECT * FROM v0; -- no row is fetched
```
I executed the query locally, with the following user name:

```sql
SELECT CURRENT_USER();
  current_user
----------------
  root
(1 row)
```

I found this issue based on commit `c76ad970e73e606bc55a372d93f9d2d6acb32c9c`.

**Comment by otan** [comment]
My suspicion is something in the constant folding: `SELECT * FROM v0 WHERE crdb_internal.no_constant_folding(CURRENT_USER() != '');`

**Comment by RaduBerinde** [comment]
Looks like a distsql planning issue. The execution plan seems fine (it does have the `c0` filter):
```
  filter         |             |
   └── limit     |             |
        │        | offset      | CAST(NULL AS INT8)
        └── scan |             |
                 | table       | t0@primary
                 | spans       | ALL
                 | filter      | c0
```
But the distsql plan is: [just this](https://cockroachdb.github.io/distsqlplan/decode.html#eJyMj01LxDAYhO_-ijiX3ZWI7TUgrGjFQt2ubUFBisTmZSl0k5oPUUr_u7Q9iAdhj_NMMg_vAPfRQSB52Wc36Y6t79KyKp-yDSuTLLmt2AW7L_JH9hmx54ekSFgTrCXt34Iju96w82u2WoFDG0U7eSQH8YoYNUdvTUPOGTuhYX6Qqi-IiKPVffATrjkaYwligG99RxCo5HtHBUlF9ioChyIv226e7W17lPZ76yde9lI7wS7BkQcv2DZGPXKY4H-3nZcHgohHfrq_INcb7eiP-r_laKw5SB1oudGZYBvaW9PMmiXm878ZKHJ-aeMlpHqpxno8-wkAAP__Tx99mw==)

**Comment by RaduBerinde** [comment]
Well, maybe it's not quite ok, it looks like it has a `filterNode` with a `nil` filter and that's throwing off the distsql planner.

**Comment by RaduBerinde** [comment]
Yeah, that's the issue, let me take this.

**Comment by RaduBerinde** [comment]
Fixed by #44307.
